### PR TITLE
Implements support for decorations for injected text.

### PR DESCRIPTION
**Pull request by hediet**
For now, only `inlineClassName` is supported.

### CODE REVIEW

**Comment by hediet** [comment]
This does not work with wrapped lines.

File: src/vs/editor/common/viewModel/viewModel.ts [hunk 1]
```diff
@@ -137,10 +141,10 @@ export class LineBreakData {
 		let delta = 0;
 		if (this.injectionOffsets !== null) {
 			for (let i = 0; i < this.injectionOffsets.length; i++) {
-				if (inputOffset < this.injectionOffsets[i]) {
+				if (inputOffset <= this.injectionOffsets[i]) {
```
Comment (alexdima) [inline src/vs/editor/common/viewModel/viewModel.ts:144]: This looks like a change unrelated to the main goal of the PR. Why was this change needed?

Was this discovered by the new test in `src/vs/editor/test/common/viewModel/lineBreakData.test.ts`?

**Review by alexdima — APPROVED** [review]
:+1: Very nice, thanks for the comments in `SplitLine.getViewLineData` which helped me understand what's going on.

File: src/vs/editor/common/viewModel/viewModel.ts (hunk 1 above)
Comment (hediet) [reply src/vs/editor/common/viewModel/viewModel.ts:144]: > Was this discovered by the new test in src/vs/editor/test/common/viewModel/lineBreakData.test.ts?

Kind of. It is part of the commit "Improves cursor positioning for injected text.".
It makes the situation a little bit better until the general solution landed.

I discovered this when adding more tests.

Comment (alexdima) [reply src/vs/editor/common/viewModel/viewModel.ts:144]: This might be one of those things that depends on the injected text being a `before` or `after` injected text.
